Sources/Datadog/FeaturesIntegration/TracingWithRUMIntegration.swift

```diff
@@ -24,41 +24,52 @@ internal struct TracingWithRUMContextIntegration {
 
 /// Sends given `Span` as RUM Errors.
 internal struct TracingWithRUMErrorsIntegration {
-    struct Attributes {
-        static let spanErrorMessage = "span.error_message"
-        static let spanErrorType = "span.error_type"
-        static let spanErrorStack = "span.error_stack"
-    }
-
     private let rumErrorsIntegration = RUMErrorsIntegration()
 
     func addError(for ddspan: DDSpan) {
-        let rumErrorMessage = "Span \"\(ddspan.operationName)\" reported an error"
-        let rumErrorAttributes = captureRUMErrorAttributesFromOTLogFields(in: ddspan)
+        let rumErrorAttributes = captureRUMErrorAttributes(from: ddspan)
 
-        rumErrorsIntegration.addError(with: rumErrorMessage, attributes: rumErrorAttributes)
+        rumErrorsIntegration.addError(
+            with: rumErrorAttributes.message,
+            stack: rumErrorAttributes.stack,
+            source: .source,
+            attributes: nil
+        )
     }
 
     /// Inspects the `DDSpan` tags set explicitly by the user with `span.setTag(key:value:)`
     /// or passed by `span.log(fields:)` using Open Tracing fields.
     ///
     /// Captures the ones describing the error context and maps them to RUM Error event attributes.
-    private func captureRUMErrorAttributesFromOTLogFields(in ddspan: DDSpan) -> [AttributeKey: AttributeValue] {
+    private func captureRUMErrorAttributes(from ddspan: DDSpan) -> (message: String, stack: String?) {
         let openTracingFieldsReducer = SpanTagsReducer(spanTags: ddspan.tags, logFields: ddspan.logFields)
         let spanTags = openTracingFieldsReducer.reducedSpanTags
 
-        var rumErrorAttributes: [AttributeKey: AttributeValue] = [:]
+        let spanErrorMessage = spanTags[DDTags.errorMessage] as? String
+        let spanErrorType = spanTags[DDTags.errorType] as? String
+        let spanErrorStack = spanTags[DDTags.errorStack] as? String
 
-        if let spanErrorMessage = spanTags[DDTags.errorMessage] {
-            rumErrorAttributes[Attributes.spanErrorMessage] = spanErrorMessage
-        }
-        if let spanErrorType = spanTags[DDTags.errorType] {
-            rumErrorAttributes[Attributes.spanErrorType] = spanErrorType
+        switch (spanErrorMessage, spanErrorType) {
+        case (let message?, let type?):
```

Comment:
oh i didn't know question marks were required here, good to know 👍 
do you know why it doesn't work without question marks?

Reply:
It's a [convenience for matching `.some(let value)`](https://docs.swift.org/swift-book/ReferenceManual/Patterns.html#ID520) on the optional type. As Swift doc says, this can be also used in loops:
```swift
let arrayOfOptionalInts: [Int?] = [nil, 2, 3, nil, 5]
// Match only non-nil values.
for case let number? in arrayOfOptionalInts {
    print("Found a \(number)")
}
```